Kernel/Bus/PCI/Device.h

```diff
@@ -33,19 +41,65 @@ struct [[gnu::packed]] MSIxTableEntry {
     u32 vector_control;
 };
 
-class Device {
+class Bus;
+class HostController;
+class Driver;
+class Device final : public AtomicRefCounted<Device> {
+    AK_MAKE_NONCOPYABLE(Device);
+
+    friend class Bus;
+    friend class HostController;
+    friend class Driver;
+    friend class ClassedDriverDeviceLists;
```

Comment:
`ClassedDriverDeviceLists` is a struct. You also need to change this to `friend struct` in Driver.h